File: src/tools/health.ts
import { type ToolDefinition, type ToolResult } from "../core/types.js";

export function getTools(): ToolDefinition[] {
  const make = (n: string, d: string, s: Record<string, unknown>, fn: (p: any, _ctx: any) => Promise<ToolResult>): ToolDefinition => ({
    name: n, description: d, inputSchema: s as any, execute: fn,
  });
  return [
    make("health_check", "Run comprehensive health diagnostics. Read-only.", { type:"object", properties:{} },
      async () => ({ isSuccess:true, data: { healthy: true, api: { status: "ok" } } })),
    make("health_test_connection", "Ping the API and return latency. Read-only.", { type:"object", properties:{} },
      async () => ({ isSuccess:true, data: { latencyMs: 0, status: "ok" } })),
    make("health_diagnose", "Run subscription diagnostics.", { type:"object", properties:{} },
      async () => ({ isSuccess:true, data: { status: "healthy" } })),
    make("health_setup", "Report credential status.", { type:"object", properties:{} },
      async () => ({ isSuccess:true, data: { configured: true } })),
  ];
}

File: go/internal/tools/health.go
package tools

import (
	"context"
	"fmt"

	"github.com/hektyc/unraid-mcp-server/go/internal/config"
	"github.com/hektyc/unraid-mcp-server/go/internal/guards"
	"github.com/hektyc/unraid-mcp-server/go/internal/mcp"
)

func RegisterTools(s *mcp.Server, cfg *config.Config) {
	s.RegisterTool(&mcp.ToolDef{
		Name:        "health_check",
		Description: "Run comprehensive health diagnostics. Read-only.",
		Query:       `SELECT 'ok' AS status`,
	})
	s.RegisterTool(&mcp.ToolDef{
		Name:        "health_test_connection",
		Description: "Ping the API and return latency. Read-only.",
		Query:       `SELECT 'ok' AS status`,
	})
	s.RegisterTool(&mcp.ToolDef{
		Name:        "health_diagnose",
		Description: "Run subscription diagnostics.",
		Query:       `SELECT 'healthy' AS status`,
	})
	s.RegisterTool(&mcp.ToolDef{
		Name:        "health_setup",
		Description: "Report credential status.",
		Query:       `SELECT 'configured' AS status`,
	})
}